src/pybind/mgr/dashboard/frontend/src/app/shared/api/mgr-module.service.ts

```diff
@@ -85,9 +86,11 @@ export class MgrModuleService {
     table: TableComponent = null,
     navigateTo: string = '',
     notificationText?: string,
-    navigateByUrl?: boolean
+    navigateByUrl?: boolean,
+    reconnectingMessage: string = $localize`Reconnecting, please wait ...`
   ): void {
     let $obs;
+
     const fnWaitUntilReconnected = () => {
```

Comment:
This function can be refactored as it may have some issues, try using RxJs retry patterns, switchMap.